python/GafferUI/UIEditor.py

```diff
@@ -196,7 +205,9 @@ def __updateFromSetInternal( self, lazy=True ) :
 			self.__sectionEditor.setPlugParent( None )
 		else :
 			plugParent = self.__node["user"]
-			if isinstance( self.__node, Gaffer.Box ) :
+			isBox = isinstance( self.__node, Gaffer.Box )
+			self.__iconRow.setVisible( isBox )
```

Comment:
We don't have this conditional visibility for other things like description, documentation URL etc. I quite like that folks can customise _any_ node in these regards, so I don't see any harm in allowing the same for the icon.